### ISSUE TITLE
pytorch blocked with two machine 

### ISSUE THREAD

**Issue by jiawensong** [horovod/horovod#1710]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
pytorch
2. Framework version:
torch==1.3.1
3. Horovod version:
0.19.0
4. MPI version:
openmpi-4.0.1
5. CUDA version:
cuda:10.0-cudnn7-devel-ubuntu18.04
6. NCCL version:
2.4.2
7. Python version:
python3.6
8. OS and version:
ubuntu18.04
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
yes
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
yes
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?
yes

**Bug report:**
Please describe errorneous behavior you're observing and steps to reproduce it.
i have run https://github.com/horovod/horovod/blob/master/examples/pytorch_imagenet_resnet50.py
in two machine,each machine use 4 gpus. The code will block at line 196:optimizer.step() when i use rtx2080ti to train, but can run successful when i use titanxp to train.
The last few lines of blocking log are usually like this：
ht-test-softmax-rtx2080ti-mpiworker-0:18:255 [1] include/shm.h:42 NCCL WARN Cuda failure 'invalid argument'
ht-test-softmax-rtx2080ti-mpiworker-0:18:255 [1] NCCL INFO transport/shm.cu:211 -> 1
ht-test-softmax-rtx2080ti-mpiworker-0:18:255 [1] NCCL INFO init.cu:667 -> 1
ht-test-softmax-rtx2080ti-mpiworker-0:18:255 [1] NCCL INFO init.cu:804 -> 1
ht-test-softmax-rtx2080ti-mpiworker-0:18:255 [1] NCCL INFO init.cu:915 -> 1
ht-test-softmax-rtx2080ti-mpiworker-0:19:262 [2] NCCL INFO comm 0x7f3ca02a2920 rank 2 nranks 8 cudaDev 2 nvmlDev 4 - Init COMPLETE
ht-test-softmax-rtx2080ti-mpiworker-0:17:266 [0] NCCL INFO Ring 00 : 0 -> 4 [send] via NET/Socket/0
ht-test-softmax-rtx2080ti-mpiworker-0:17:266 [0] NCCL INFO Trees [0] -1->0->1/4/-1
ht-test-softmax-rtx2080ti-mpiworker-0:17:266 [0] NCCL INFO Using 256 threads, Min Comp Cap 7, Trees enabled up to size 149999
ht-test-softmax-rtx2080ti-mpiworker-1:15:252 [0] NCCL INFO Trees [0] 0->4->5/-1/-1
ht-test-softmax-rtx2080ti-mpiworker-1:15:252 [0] NCCL INFO comm 0x7f27c0252c00 rank 4 nranks 8 cudaDev 0 nvmlDev 0 - Init COMPLETE
ht-test-softmax-rtx2080ti-mpiworker-0:17:266 [0] NCCL INFO comm 0x7fd67c28b850 rank 0 nranks 8 cudaDev 0 nvmlDev 2 - Init COMPLETE

**Comment by tgaddair** [comment]
Hey @jiawensong, what version of NCCL are you using?  We've seen this error come up a few times before (#893, #1122), and in both cases it was caused by a bug in NCCL 2.4.X.

**Comment by jiawensong** [comment]
> Hey @jiawensong, what version of NCCL are you using? We've seen this error come up a few times before (#893, #1122), and in both cases it was caused by a bug in NCCL 2.4.X.

i meet this problem in nccl 2.4.2. Then i have changed the version to 2.5.6, the code can run correctly. Thanks for your help.
